pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -157,16 +163,23 @@ func (r *hibernationReconciler) Reconcile(request reconcile.Request) (result rec
 	shouldHibernate := cd.Spec.PowerState == hivev1.HibernatingClusterPowerState
 	hibernatingCondition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.ClusterHibernatingCondition)
 
-	// Signal a problem if we should be hibernating or have requested hibernate after and the cluster does not support it.
+	// Signal a problem if we should be hibernating or have requested hibernate after and the cluster does not support it or
+	// SyncSets have not yet been applied.
 	if shouldHibernate || cd.Spec.HibernateAfter != nil {
-		if supported, msg := r.canHibernate(cd); !supported {
+		if supported, msg := r.hibernationSupported(cd); !supported {
 			return r.setHibernatingCondition(cd, hivev1.UnsupportedHibernationReason, msg, corev1.ConditionFalse, cdLog)
 		}
+		if syncSetsApplied, msg := r.syncSetsApplied(cd); !syncSetsApplied {
+			// Allow hibernation if hibernateAfterSyncSetsNotApplied duration has passed since cluster installed and syncsets still not applied
+			if cd.Status.InstalledTimestamp != nil && time.Now().Sub(cd.Status.InstalledTimestamp.Time) < hibernateAfterSyncSetsNotApplied {
+				return r.setHibernatingCondition(cd, hivev1.SyncSetsNotAppliedReason, msg, corev1.ConditionFalse, cdLog)
+			}
```

Comment:
When this is not the case and we've past expiry, is there any indication on the cluster deployment that we were not able to apply syncsets even though hibernating is now true? I suspect there probably is, via our pre-existing syncsets failing condition. If so this is great.

Reply:
We'll have a `SyncSetApplyFailure` failing condition, right.

```  - lastProbeTime: "2020-11-18T15:40:04Z"
    lastTransitionTime: "2020-11-18T15:40:04Z"
    message: One of the SyncSet applies has failed
    reason: SyncSetApplyFailure
    status: "True"
    type: SyncSetFailed
  - lastProbeTime: "2020-11-18T15:40:04Z"
    lastTransitionTime: "2020-11-18T15:40:02Z"
    message: Cluster SyncSets have not been applied
    reason: SyncSetsNotApplied
    status: "False"```